core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -194,7 +196,12 @@ public String getLocation(@Nullable String routing) {
         if (routing != null) {
             location.append(routingStart).append(routing);
         }
-        return location.toString();
+        try {
+            URI uri = new URI(location.toString());
+            return uri.toASCIIString();
+        } catch (URISyntaxException ex) {
```

Comment:
I'm fairly sure `Location` is supposed to be a valid URI. I'd be ok with catching the exception and logging a warning for it unless we expect this to fail frequently. If we *do* then we shouldn't be using `URI`.